internal/distro/distro_test.go

```diff
@@ -145,3 +148,96 @@ func TestImageType_PackageSetsChains(t *testing.T) {
 		}
 	}
 }
+
+// Ensure all image types report the correct names for their pipelines.
+// Each image type contains a list of build and payload pipelines. They are
+// needed for knowing the names of pipelines from the static object without
+// having access to a manifest, which we need when parsing metadata from build
+// results.
+func TestImageTypePipelineNames(t *testing.T) {
+	// types for parsing the opaque manifest with just the fields we care about
+	type pipeline struct {
+		Name string `json:"name"`
+	}
+	type manifest struct {
+		Pipelines []pipeline `json:"pipelines"`
+	}
+
+	require := require.New(t)
+	distros := distroregistry.NewDefault()
+	for _, distroName := range distros.List() {
+		d := distros.GetDistro(distroName)
+		for _, archName := range d.ListArches() {
+			arch, err := d.GetArch(archName)
+			require.Nil(err)
+			for _, imageTypeName := range arch.ListImageTypes() {
+				t.Run(fmt.Sprintf("%s/%s/%s", distroName, archName, imageTypeName), func(t *testing.T) {
+					imageType, err := arch.GetImageType(imageTypeName)
+					require.Nil(err)
+
+					// set up bare minimum args for image type
+					customizations := &blueprint.Customizations{}
+					if imageType.Name() == "edge-simplified-installer" {
+						customizations = &blueprint.Customizations{
+							InstallationDevice: "/dev/null",
+						}
+					}
+					bp := blueprint.Blueprint{
+						Customizations: customizations,
+					}
+					options := distro.ImageOptions{}
+					repos := make([]rpmmd.RepoConfig, 0)
+					containers := make([]container.Spec, 0)
+					seed := int64(0)
+
+					// Add ostree options for image types that require them
+					options.OSTree = distro.OSTreeImageOptions{
+						ImageRef:      imageType.OSTreeRef(),
+						URL:           "https://example.com/repo",
```

Comment:
I like using `example.com` because that's what it's there for.  Clearly broken can be messy.  Sometimes something that is clearly broken becomes valid down the line or we might add some validation that breaks the test.  The latter is a smaller concern, we could just fix the test with a new url when it breaks.